src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeRelation.scala

```diff
@@ -142,4 +143,95 @@ class DeltaLakeRelation(spark: SparkSession, override val plan: LogicalRelation)
       /* modificationTime */ modificationTime,
       path)
   }
+
+  private def deltaLakeVersionHistory(index: IndexLogEntry): Seq[(Int, Int)] = {
+    val versions =
+      index.derivedDataset.properties.properties
+        .getOrElse(DeltaLakeConstants.DELTA_VERSION_HISTORY_PROPERTY, "")
+
+    // The value is comma separated versions - <index log version>:<delta table version>.
+    // e.g. "1:2,3:5,5:9"
+    versions.split(",").reverseIterator.foldLeft(Seq[(Int, Int)]()) { (list, versionStr) =>
+      val pair = versionStr.split(":")
+      val Array(indexLogVersion, deltaTableVersion) = versionStr.split(":").map(_.toInt)
+      if (list.nonEmpty && list.head._2 == deltaTableVersion) {
+        // Omit if the delta lake version is the same. In this case, we only take the higher
+        // index log version as this is usually from index optimizations.
```

Comment:
"usually" or "always"? If "usually", what are the other scenarios?

Reply:
It's the only case; removed "usually".